state/execution.go

```diff
@@ -147,7 +147,7 @@ func updateValidators(validators *types.ValidatorSet, changedValidators []*tmsp.
 		}
 
 		address := pubkey.Address()
-		power := int64(v.Power)
+		power := uint64(v.Power)
```

Comment:
This is the check @ebuchman was referring to I think.  We make sure they can't overflow when the TMSP app sends a validator set update.  This worked for int64 or int32, since it is clear when there is a negative value.

Let's say the tmsp app has a tx that reduces a voting power by 2.  And it doesn't check underflow.  If the user was already at 1, they would do the bad math in the app => math.Maxunit64, which is sent to tendermint core, which cannot reliably distinguish if there was an error in the app, or this validator is suddenly REALLY important.

The old code at least would catch and report errors in app code. But I do agree, internally it is a bit annoying to have ints when you want uints...

Not sure the solution, but core should be defensive against errors in tmsp apps.